business_logic/db/DBCommunicator.js

```diff
@@ -14,42 +14,56 @@ experimentsCollection.loadExperimentsCollection(database_global)
 actionsCollection.loadActionsCollection(database_global)
 tweetsCollection.loadTweetsCollection(database_global)
 
+//TO BE CHANGED AFTER HACKHATON!!
 function insertExperiment (experiment){
-    experimentsCollection.insertExperiment(experiment)
-    // Call for more collection modules if needed.
+    experimentsCollection.insertExperiment(experiment);
+    //wipe the whole db
+    actionsCollection.deleteActions();
+    participantsCollection.deleteParticipants();
+    researchersCollection.deleteResearchers();
+    tweetsCollection.deleteTweets();
 }
 
 function getExperimentByCode(expCode){
+    return experimentsCollection.getExperimentByCode(expCode);
 
 }
 
 function insertParticipant(participant) {
-    // Insert to experimentsCollection
-    // Insert to participantsCollection
+    var parsedData = JSON.parse(participant);
+    var expId;
+    for(item in parsedData) {
+        if(item=="exp_id"){
+            expId=parsedData[item];
+        }
+    }
```

Comment:
What did you try to do here? Maybe you meant: `expId = participant.exp_id` instead of all those lines?

Reply:
fixed